R/check_crpr.R

```diff
@@ -0,0 +1,131 @@
+.datasets <- new.env(parent = emptyenv())
+
+#' Signal CR Records Followed by PR
+#'
+#' @param dataset A data frame
+#'
+#' @param order A list of variables created using `vars()` determining the order
+#'   or the records
+#'
+#' @param msg The condition message
+#'
+#' @param subject_keys Variables to uniquely identify a subject
+#'
+#'   A list of symbols created using `vars()` is expected.
+#'
+#' @param check_type Type of message to issue when detecting PR after CR.
+#'
+#'   *Default*: `"warning"`
+#'
+#'   *Permitted Values*: `"warning"` or `"error"`
+#'
+#' @return No return value, called for side effects
+#'
+#' @seealso [get_crpr_dataset()]
+#'
+#' @export
+#' @keywords user_utility
+#' @author Stefan Bundfuss
+#'
+#' @examples
+#' library(tibble)
+#' library(dplyr)
+#' library(lubridate)
+#' library(admiralonco)
+#'
+#' adrs <- tribble(
+#'   ~USUBJID, ~ADTC,        ~AVALC,
+#'   "1",      "2020-01-01", "PR",
+#'   "1",      "2020-02-01", "CR",
+#'   "1",      "2020-02-16", "NE",
+#'   "1",      "2020-03-01", "CR",
+#'   "2",      "2020-02-06", "PR",
+#'   "2",      "2020-02-16", "CR",
+#'   "2",      "2020-03-30", "PR",
+#' ) %>%
+#'   mutate(
+#'     ADT = ymd(ADTC),
+#'     STUDYID = "XX1234"
+#'   )
+#'
+#' signal_crpr(adrs, order = vars(ADT))
+signal_crpr <- function(dataset,
+                        order,
+                        msg = "Dataset contains CR records followed by PR.",
+                        subject_keys = vars(STUDYID, USUBJID),
+                        check_type = "warning") {
+  assert_order_vars(order)
+  assert_vars(subject_keys)
+  assert_data_frame(dataset, required_vars = quo_c(order, subject_keys))
```

Comment:
Yes, you are right. According to the programming strategy we should not check `order` and `dataset`. I have removed the assertions.

I would not change `subject_keys` to `by_vars` or vive versa because `filter_confrimation()` is much more general than `signal_crpr()`.